Mathlib/RingTheory/RingHom/StandardSmooth.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.RingTheory.LocalProperties.Basic
+import Mathlib.RingTheory.Smooth.StandardSmooth
+
+/-!
+# Standard smooth ring homomorphisms
+
+In this file we define standard smooth ring homomorphisms and show their
+meta properties.
+
+## Notes
+
+This contribution was created as part of the AIM workshop "Formalizing algebraic geometry"
+in June 2024.
+
+-/
+
+universe t t' w w' u v
+
+variable (n m : ℕ)
+
+open TensorProduct
+
+namespace RingHom
+
+variable {R : Type u} {S : Type v} [CommRing R] [CommRing S]
+
+/-- A ring homomorphism `R →+* S` is standard smooth if `S` is standard smooth as `R`-algebra. -/
+def IsStandardSmooth (f : R →+* S) : Prop :=
+  @Algebra.IsStandardSmooth.{t, w} _ _ _ _ f.toAlgebra
+
+/-- A ring homomorphism `R →+* S` is standard smooth of relative dimension `n` if
+`S` is standard smooth of relative dimension `n` as `R`-algebra. -/
+def IsStandardSmoothOfRelativeDimension (f : R →+* S) : Prop :=
+  @Algebra.IsStandardSmoothOfRelativeDimension.{t, w} n _ _ _ _ f.toAlgebra
+
+lemma IsStandardSmoothOfRelativeDimension.isStandardSmooth (f : R →+* S)
+    (hf : IsStandardSmoothOfRelativeDimension.{t, w} n f) :
+    IsStandardSmooth.{t, w} f :=
+  letI : Algebra R S := f.toAlgebra
+  letI : Algebra.IsStandardSmoothOfRelativeDimension.{t, w} n R S := hf
```

Comment:
Can we now use `algebraize` here and below?

Reply:
I would like to, but as far as I can tell `algebraize` has not landed yet, am I missing something (#16217 is still open).

Reply:
Oooh, I thought it had already landed! I kicked it on the queue. Please merge master in a few moments.

Reply:
Voila: `algebraize` is available

Reply:
`algebraize` is in use now. Unfortunately for `IsStandardSmooth` things don't work as smoothly (...) as I had hoped: Using `algebraize` with `IsStandardSmooth` tagged with `@[algebraize]` causes a metavariable error. I suspect this is because `Algebra.IsStandardSmooth` depends on two additional universe levels. Also for `IsStandardSmoothOfRelativeDimension n` it does likely not work because of the `n` parameter.

Reply:
@callesonne Do you have an idea whether this would be easy to fix?

Reply:
Let's return to this at a later point, when `algebraize` has been extended.